app/src/main/java/org/dharmaseed/android/TalkRepository.java

```diff
@@ -31,37 +31,42 @@ public Cursor getTalkAdapterData(
             boolean isDownloaded
     )
     {
-        List<String> columns = new ArrayList<>();
-        columns.add(DBManager.C.Talk.TABLE_NAME    + "." + DBManager.C.Talk.ID);
-        columns.add(DBManager.C.Talk.TABLE_NAME    + "." + DBManager.C.Talk.TITLE);
-        columns.add(DBManager.C.Talk.TABLE_NAME    + "." + DBManager.C.Talk.TEACHER_ID);
-        columns.add(DBManager.C.Teacher.TABLE_NAME + "." + DBManager.C.Teacher.NAME);
-        return getTalks(columns, searchTerms, isStarred, isDownloaded);
+
+        return getTalks(searchTerms, isStarred, isDownloaded, "");
     }
 
     /**
-     * @param columns the columns to select or null if all
      * @param searchTerms the term to filter by or null if none
      * @param isStarred whether the talk is starred
      * @param isDownloaded whether the talk is downloaded
+     * @param extraWhere an extra where clause that can be used to filter on a specific teacher/center id
      * @return every talk in the db filtered by search term, is starred, and is downloaded
      */
     private Cursor getTalks(
-           List<String> columns,
            List<String> searchTerms,
            boolean isStarred,
-           boolean isDownloaded
+           boolean isDownloaded,
+           String extraWhere
     )
     {
+        String[][] columns = {
+            {DBManager.C.Talk.TABLE_NAME, DBManager.C.Talk.TITLE},
+            {DBManager.C.Talk.TABLE_NAME, DBManager.C.Talk.TEACHER_ID},
+            {DBManager.C.Teacher.TABLE_NAME, DBManager.C.Teacher.NAME},
+            {DBManager.C.Center.TABLE_NAME, DBManager.C.Center.NAME},
+            {DBManager.C.Talk.TABLE_NAME, DBManager.C.Talk.DURATION_IN_MINUTES},
+            {DBManager.C.Talk.TABLE_NAME, DBManager.C.Talk.RECORDING_DATE},
+        };
+
         String queryColumns = "";
-        if (columns == null || columns.size() == 0)
-        {
-            queryColumns = "*";
-        }
-        else
-        {
-            queryColumns = TextUtils.join(", ", columns);
+        List<String> named_columns = new ArrayList<String>();
+        named_columns.add(DBManager.C.Talk.TABLE_NAME + "." + DBManager.C.Talk.ID);
```

Comment:
remove this line and only use the columns passed in. Besides, we'd want to alias this too, because `_id` could conflict for talks, teachers, and centers.

Reply:
Yeah, I really don't like treating this column differently either. :/  However, since we're using [CursorAdapter](https://developer.android.com/reference/android/widget/CursorAdapter), the query must return a column that is called `_id` exactly.  Here's what happens if you try to alias it:

`java.lang.RuntimeException: Unable to resume activity {org.dharmaseed.android/org.dharmaseed.android.NavigationActivity}: java.lang.IllegalArgumentException: column '_id' does not exist. Available columns: [talks__id, talks_title, talks_teacher_id, teachers_name, centers_name, talks_duration_in_minutes, talks_rec_date]`

So it seems like some special treatment of this column is required, as far as I can tell.  Let me know if you can think of any clever ways around this though. :)